sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateInitialStateSuite.scala

```diff
@@ -395,6 +395,10 @@ class TransformWithStateInitialStateSuite extends StateStoreMetricsTest
         AddData(inputData, InitInputRow("k2", "update", 40.0)),
         AddData(inputData, InitInputRow("non-exist", "getOption", -1.0)),
         CheckNewAnswer(("non-exist", "getOption", -1.0)),
+        Execute { q =>
+          assert(q.lastProgress
```

Comment:
This assertion seems flaky:
- https://github.com/apache/spark/actions/runs/12069501936/job/33656903689
- https://github.com/apache/spark/actions/runs/12060929700/job/33632163805

![image](https://github.com/user-attachments/assets/9d9eef75-3594-4a5d-a772-2e363eddeef9)

Reply:
The new failed link i n change pipeline:
- https://github.com/apache/spark/actions/runs/12192978771/job/34014487080
- https://github.com/apache/spark/actions/runs/12191406660/job/34010259589
- https://github.com/apache/spark/actions/runs/12185951744/job/33993383845
- https://github.com/apache/spark/actions/runs/12164293819/job/33925652391

![image](https://github.com/user-attachments/assets/a3fb09f9-0fcf-44c6-97a8-06db832b7f09)